concent_api/core/message_handlers.py

```diff
@@ -956,34 +957,33 @@ def store_subtask(
 def handle_messages_from_database(client_public_key: bytes) -> Union[message.Message, None]:
     assert client_public_key not in ['', None]
     encoded_client_public_key = b64encode(client_public_key)
-
     with non_nesting_atomic(using='control'):
-        pending_response = PendingResponse.objects.select_for_update().filter(
+        payments_info_responses = PendingResponse.objects.select_for_update().filter(
+            payment_info__isnull=False,
             client__public_key=encoded_client_public_key,
             delivered=False,
         ).order_by('created_at').first()
 
-        if pending_response is None:
-            return None
+        subtask_responses = PendingResponse.objects.select_for_update().filter(
+            subtask__protocol_version=get_major_and_minor_golem_messages_version(settings.GOLEM_MESSAGES_VERSION),
+            client__public_key=encoded_client_public_key,
+            delivered=False,
+        ).order_by('created_at').first()
+        if payments_info_responses is not None and subtask_responses is not None:
+            if payments_info_responses.created_at > subtask_responses.created_at:
+                pending_response = payments_info_responses
+            else:
+                pending_response = subtask_responses
+        else:
+            if payments_info_responses is not None:
```

Comment:
Can be simplified to 
`pending_response = payments_info_responses or subtask_responses`

Reply:
Done